src/AbstractRole.php

```diff
@@ -103,16 +103,33 @@ public function addChild($child)
      */
     public function setParent($parent)
     {
-        $this->parent = $parent;
-
+        if (null === $this->parents) {
+            $this->parents = [];
+        }
+        if (!in_array($parent, $this->parents)) {
```

Comment:
Should this add the boolean third argument to `in_array()`? Also wondering if it should check against the `(string)` value of each parent, as each could be a `RoleInterface`?

Reply:
I would like to refactor this approach for ver. 3.0.0 and use only `RoleInterface` as valid input parameter. Because we are adding `addParent()` already in ver. 2.6.0, we can start using this approach. I'll check for `RoleInterface` instance now and we can use type hinting for the future.